File: benchmarks/prime/prime.go
package main

import "fmt"

func main() {
	x := getPrimes(25_000)
	fmt.Println("Go:", len(x))
}

func getPrimes(n int) []int {
	primes := []int{}

	for i := 2; i < n; i++ {
		if isPrime(i) {
			primes = append(primes, i)
		}
	}

	return primes
}

func isPrime(n int) bool {
	for i := 2; i < n; i++ {
		if n%i == 0 {
			return false
		}
	}

	return true
}

File: benchmarks/prime/prime.js
function getPrimes(n) {
  let primes = [];

  for (let i = 2; i < n; i++) {
    if (isPrime(i)) {
      primes.push(i);
    }
  }

  return primes;
}

function isPrime(n) {
  for (let i = 2; i < n; i++) {
    if (n % i == 0) {
      return false;
    }
  }

  return true;
}

let x = getPrimes(25_000);
console.log("Javascript:", x.length);
